qiskit/circuit/library/arithmetic/piecewise_polynomial_pauli_rotations.py

```diff
@@ -0,0 +1,260 @@
+        """
+        return self._breakpoints
+
+    @breakpoints.setter
+    def breakpoints(self, breakpoints: List[int]) -> None:
+        """Set the breakpoints.
+
+        Args:
+            breakpoints: The new breakpoints.
+        """
+        self._invalidate()
+        self._breakpoints = breakpoints
+
+        if self.num_state_qubits and breakpoints:
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def coeffs(self) -> List[List[float]]:
+        """The coefficients of the polynomials.
+
+        The function is polynomial in the intervals ``[point_i, point_{i+1}]`` where the last
+        point implicitly is ``2**(num_state_qubits + 1)``.
+        """
+        return self._coeffs
+
+    @coeffs.setter
+    def coeffs(self, coeffs: List[List[float]]) -> None:
+        """Set the polynomials.
+
+        Args:
+            coeffs: The new polynomials.
+        """
+        self._invalidate()
+        self._coeffs = coeffs
+
+        # update the homogeneous polynomials and degree
+        self._hom_coeffs = []
+        self._degree = len(max(self._coeffs, key=len)) - 1
+        for poly in self._coeffs:
+            self._hom_coeffs.append(poly + [0] * (self._degree + 1 - len(poly)))
+
+    @property
+    def mapped_coeffs(self) -> List[List[float]]:
+        """The coefficients mapped to the internal representation, since we only compare
+        x>=breakpoint.
+
+        Returns:
+            The mapped coefficients.
+        """
+        mapped_coeffs = []
+
+        # First polynomial
+        mapped_coeffs.append(self._hom_coeffs[0])
+        for i in range(1, len(self._hom_coeffs)):
+            mapped_coeffs.append([])
+            for j in range(0, self._degree+1):
+                mapped_coeffs[i].append(self._hom_coeffs[i][j] - self._hom_coeffs[i-1][j])
+
+        return mapped_coeffs
+
+    @property
+    def contains_zero_breakpoint(self) -> bool:
+        """Whether 0 is the first breakpoint.
+
+        Returns:
+            True, if 0 is the first breakpoint, otherwise False.
+        """
+        return np.isclose(0, self.breakpoints[0])
+
+    def evaluate(self, x: float) -> float:
+        """Classically evaluate the piecewise polynomial rotation.
+
+        Args:
+            x: Value to be evaluated at.
+
+        Returns:
+            Value of piecewise polynomial function at x.
+        """
+
+        y = 0
+        for i in range(0, len(self.breakpoints)):
+            y = y + (x >= self.breakpoints[i]) * (np.poly1d(self.mapped_coeffs[i][::-1])(x))
+
+        return y
+
+    @property
+    def num_ancilla_qubits(self) -> int:
+        """The number of ancilla qubits.
+
+        Returns:
+            The number of ancilla qubits in the circuit.
+        """
+        num_ancilla_qubits = self.num_state_qubits - 1 + len(self.breakpoints)
+        if self.contains_zero_breakpoint:
+            num_ancilla_qubits -= 1
+        return num_ancilla_qubits + max(1, self._degree - 1)
```

Comment:
Same as for Chebychev code: this should not be necessary. To be more precise, you can put this calculation of number of ancillas into `_reset_registers` and then add a `AncillaRegister` of appropriate size